sqlmesh/dbt/context.py

```diff
@@ -52,8 +50,8 @@ class DbtContext:
     _manifest: t.Optional[ManifestHelper] = None
 
     @property
-    def dialect(self) -> t.Optional[str]:
-        return self.model_defaults.dialect if self.model_defaults else None
+    def dialect(self) -> str:
+        return self.engine_adapter.dialect if self.engine_adapter is not None else ""
```

Comment:
we should raise here right? why would engine_adapter be none

Reply:
Raising makes sense to me, but these lines are a straight reversion to previous so I'm not totally sure. @crericha

Reply:
The optional nature of the adapter may not be needed any longer. @izeigerman ?

Reply:
Actually I don't think dbt's context even need an engine adapter at this point. Why don't we just look at the target instead and raise if one is not available, since it should definitely be available at that point.

Reply:
Removed engine adapter from context